apps/mobile/src/utils/version-utils.ts

```diff
@@ -0,0 +1,89 @@
+/**
+ * Utility functions for version comparison and validation
+ */
+
+export interface SemanticVersion {
+  major: number;
+  minor: number;
+  patch: number;
+}
+
+/**
+ * Parse a semantic version string into components
+ * @param version - Version string (e.g., "2.57.0")
+ * @returns Parsed version components
+ * @throws Error if version format is invalid
+ */
+export function parseSemanticVersion(version: string): SemanticVersion {
+  if (!version || typeof version !== 'string') {
+    throw new Error('Version must be a non-empty string');
+  }
+
+  const parts = version.split('.');
+  if (parts.length < 2 || parts.length > 3) {
+    throw new Error('Version must be in format x.y or x.y.z');
+  }
+
+  const major = parseInt(parts[0] || '0', 10);
+  const minor = parseInt(parts[1] || '0', 10);
+  const patch = parts[2] ? parseInt(parts[2], 10) : 0;
+
+  if (isNaN(major) || isNaN(minor) || isNaN(patch)) {
+    throw new Error('Version components must be numeric');
```

Comment:
Can this be made resilient without throwing errors? I'd love to see a more idiomatic control flow that doesn't depend on try/catch, either by relying on fallback, or a union type returning {valid: false} | {valid: true: result: ...}